src/ax_interface/mib.py

```diff
@@ -228,13 +240,21 @@ def __init__(self, mib_cls, update_frequency=DEFAULT_UPDATE_FREQUENCY):
         self.updater_instances = getattr(mib_cls, MIBMeta.UPDATERS)
         self.prefixes = getattr(mib_cls, MIBMeta.PREFIXES)
 
+    def done_background_task_callback(fut):
+        ex = fut.exception()
+        if ex is not None:
+            exstr = "MIBTable background task caught an unexpected exception: {}".format(str(ex))
+            logger.error(exstr)
```

Comment:
I don't use 'logger.exception' here on purpose. All the updater's exceptions or errors should be caught by MIBUpdater, if anything caught here, it's really a logic error in the MIBUpdater or MIBTable itself, which is truly unexpected.

---
In reply to: [121222352](https://github.com/Azure/sonic-snmpagent/pull/32#discussion_r121222352) [](ancestors = 121222352)